app/services/PageService.scala

```diff
@@ -3,32 +3,49 @@ package services
 import java.io.{File, FileInputStream}
 
 import com.typesafe.config.Config
+import controllers.dto.PageFragment
 import javax.inject.Inject
-import models._
+import models.{PageJoinProject, _}
 import org.apache.commons.io.FileUtils
 import play.api.libs.json.Json
 import repositories._
 import utils._
 
 import scala.util.Try
 
-case class PageMeta(label: Option[String], description: Option[String])
+case class PageMeta(label: Option[String] = None, description: Option[String] = None)
 
-case class DirectoryMeta(label: Option[String], description: Option[String], pages: Option[Seq[String]], children: Option[Seq[String]])
+case class DirectoryMeta(label: Option[String] = None, description: Option[String] = None, pages: Option[Seq[String]] = None, children: Option[Seq[String]] = None)
 
-case class Meta(directory: Option[DirectoryMeta], page: Option[PageMeta])
+case class TagsModule(tags: Seq[String])
+
+case class ScenariosModule(project: Option[String] = None, branchName: Option[String] = None, feature: Option[String] = None, select: Option[TagsModule] = None)
+
+case class Module(directory: Option[DirectoryMeta] = None, page: Option[PageMeta] = None, scenarios: Option[ScenariosModule] = None)
+
+case class PageFragmentUnderProcessing(data: Option[String] = None, markdown: Option[String] = None, scenariosModule: Option[ScenariosModule] = None, scenarios: Option[Feature] = None)
+
+class PageService @Inject()(config: Config, projectRepository: ProjectRepository, directoryRepository: DirectoryRepository, pageRepository: PageRepository, gherkinRepository: GherkinRepository) {
 
 
-class PageService @Inject()(config: Config, directoryRepository: DirectoryRepository, pageRepository: PageRepository) {
-  val projectsRootDirectory = config.getString("projects.root.directory")
-  val documentationMetaFile = config.getString("documentation.meta.file")
   implicit val pageMetaFormat = Json.format[PageMeta]
   implicit val directoryMetaFormat = Json.format[DirectoryMeta]
-  implicit val metaFormat = Json.format[Meta]
 
-  val metaRegex = """\`\`\`thegardener([\s\S]*"page"[^`]*)\`\`\`""".r
+  implicit val tagsModuleFormat = Json.format[TagsModule]
+  implicit val scenariosModuleMetaFormat = Json.format[ScenariosModule]
+  implicit val metaFormat = Json.format[Module]
+
+
+  val projectsRootDirectory = config.getString("projects.root.directory")
+  val documentationMetaFile = config.getString("documentation.meta.file")
+  val baseUrl = config.getString("application.baseUrl")
+
+  val moduleStart = "```thegardener"
+  val moduleEnd = "```"
+  val pageModuleRegex = """\`\`\`thegardener([\s\S]*"page"[^`]*)\`\`\`""".r
   val imageRegex = """\!\[.*\]\((.*)\)""".r
   val referenceRegex = """\[.*\]\:\s(\S*)""".r
+  val scenariosModuleRegex = """\`\`\`thegardener([\s\S]*"scenarios"[^`]*)\`\`\`""".r
```

Comment:
When I tried to work with regex on the front, it appears to be easier to parse page without regex (just looking for ```thegardener pattern). You could have a look [here](https://github.com/KelkooGroup/theGardener/blob/61245671172416ae241044c88e986a831aca385b/frontend/src/app/_services/markdown-parser.service.ts). It also includes handling escaping sequence